scripts/conversion_script.py

```diff
@@ -574,34 +589,20 @@ def main():
         installed, transaction_id = install_convert2rhel()
         if installed:
             YUM_TRANSACTIONS_TO_UNDO.add(transaction_id)
-        run_convert2rhel()
-
-        # Gather JSON & Textual report
-        data = gather_json_report()
-        output.report = gather_textual_report()
-
-        highest_level = find_highest_report_level(actions=data["actions"])
-        # Set the first position of the list as being the final status, that's
-        # needed because `find_highest_report_level` will sort out the list
-        # with the highest priority first.
-        output.status = highest_level
 
-        # Generate report message and transform the raw data into entries for
-        # Insights.
-        output.message, output.alert = generate_report_message(highest_level)
+        stdout, returncode = run_convert2rhel()
 
-        if "successfully" in output.message:
-            gpg_key_file.keep = True
-
-            # NOTE: When c2r statistics on insights are not reliant on rpm being installed
-            # remove below line (=decide only based on install_convert2rhel() result)
-            YUM_TRANSACTIONS_TO_UNDO.remove(transaction_id)
-            # NOTE: Keep always because added/updated pkg is also kept
-            # (if repo existed, the .backup file will remain on system)
-            c2r_repo.keep = True
+        if returncode != 0:
+            output.message = (
+                "An error occurred during the conversion execution. For details, refer to "
+                "the convert2rhel log file on the host at /var/log/convert2rhel/convert2rhel.log"
+            )
+            output.report = (
+                "convert2rhel execution exited with code %s and output: %s."
+                % (returncode, stdout.rstrip("\n"))
+            )
+            return
```

Comment:
Won't this return just return from main function without actually doing what is in finally block = no output from script?

Reply:
Because there is the finally block, we will drop to that part instead. If we don't have a finally block, we would leave the try-except scope, but since we have, this is a way to return and go to the finally.

Reply:
ah right, true